flang/lib/Lower/Runtime.cpp

```diff
@@ -348,18 +348,31 @@ void Fortran::lower::genSystemClock(fir::FirOpBuilder &builder,
                                     mlir::Location loc, mlir::Value count,
                                     mlir::Value rate, mlir::Value max) {
   auto makeCall = [&](mlir::FuncOp func, mlir::Value arg) {
+    mlir::Type type = arg.getType();
+    fir::IfOp ifOp{};
+    if (type.dyn_cast<fir::PointerType>() || type.dyn_cast<fir::HeapType>()) {
+      // Check for a disassociated pointer or an unallocated allocatable.
+      mlir::IntegerType i64Ty = builder.getIntegerType(64);
+      auto addr = builder.createConvert(loc, i64Ty, arg);
+      auto zero = builder.createIntegerConstant(loc, i64Ty, 0);
```

Comment:
There is `builder.genIsNull` method, I think you should use that here instead so that we can centralize the places that are assuiming that the pointer width is 64 bits.

Reply:
I've renamed `genIsNull` to `genIsNullAddr`, and `genIsNotNull` to `genIsNotNullAddr` and switched to calling `genisNotNullAddr`.